templates/tfengine/components/terraform/states/main.tf

```diff
@@ -0,0 +1,23 @@
+{{- /* Copyright 2020 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */ -}}
+
+{{range .states -}}
+data "terraform_remote_state" "{{resourceName . "prefix"}}" {
+  backend = "gcs"
+  config = {
+    bucket = "{{$.state_bucket}}"
```

Comment:
do we want to support the case where state bucket is different?

Reply:
Maybe when there is use case for that? At least for now, when using DPT, the state bucket is set at the top level config and fully managed by DPT.